### PR TITLE
Add unit tests for #584

### PR DESCRIPTION
**Pull request by chewong**
This PR adds unit tests for changes made in #584.  It also fixes two small typos in the codebase.

### CODE REVIEW

**Comment by chewong** [comment]
I signed it

File: node/pod.go [hunk 1]
```diff
@@ -267,15 +267,28 @@ func (pc *PodController) updatePodStatus(ctx context.Context, pod *corev1.Pod) e
 			pod.Status.Reason = "NotFound"
 			pod.Status.Message = "The pod status was not found and may have been deleted from the provider"
 			for i, c := range pod.Status.ContainerStatuses {
+				var startedAt metav1.Time
+				var fininshedAt metav1.Time
+				if c.State.Waiting != nil {
+					// Skip waiting state
+					continue
```
Comment (sargun) [inline node/pod.go]: Doesn't this result in a behaviour change for waiting containers? Also, what's going on that results in this crash?

Comment (chewong) [reply node/pod.go]: #583 describes the crash in detail. `c.State.Running` is `nil` when one of the containers is in waiting state, causing a crash when VK tries to access `c.State.Running.StartedAt`. What type of behavior change are you expecting from this change?

**Comment by sargun** [comment]
I'm wondering if this function makes sense to keep longer term. Will this continue to exist if we move everything to NotifyPods?

File: node/pod.go [hunk 2]
```diff
@@ -267,15 +267,28 @@ func (pc *PodController) updatePodStatus(ctx context.Context, pod *corev1.Pod) e
...
+				} else if c.State.Terminated != nil {
+					startedAt = c.State.Terminated.StartedAt
+					fininshedAt = c.State.Terminated.FinishedAt
+				} else {
+					startedAt = c.State.Running.StartedAt
+					fininshedAt = metav1.NewTime(time.Now())
```
Comment (sargun) [inline node/pod.go]: I suggest capturing now outside the loop.

**Comment by sargun** [comment]
@chewong I think https://github.com/virtual-kubelet/virtual-kubelet/pull/751 obsoletes this.  Depending on how we proceed, that might make more sense.

**Comment by sargun** [comment]
Can you add a lifecycle test for this?

**Pull request merged**